(AAA) Access for Enterprises guide (#36600)
DataDog/documentation: content/en/getting_started/access_for_enterprises/example_implementations.md

---
title: Example Implementations
description: Enterprise access control implementation templates for four common organizational archetypes.
further_reading:
- link: "/account_management/rbac/"
  tag: "Documentation"
  text: "Access Control (RBAC)"
- link: "/account_management/rbac/data_access"
  tag: "Documentation"
  text: "Data Access Control"
- link: "/account_management/rbac/granular_access"
  tag: "Documentation"
  text: "Granular Access Control"
- link: "/getting_started/teams/"
  tag: "Documentation"
  text: "Getting Started with Teams"
- link: "https://registry.terraform.io/providers/DataDog/datadog/latest/docs"
  tag: "Documentation"
  text: "Datadog Terraform Provider"
---

## Overview

This section provides four implementation templates for Datadog access controls, each representing the end state for a different enterprise archetype. They show what a fully implemented access strategy looks like: which mechanisms are in use, how roles and teams are structured, and how the layers work together.

> These templates are composites, not descriptions of specific companies. Use them as starting points to design your own access implementation.

Reviewer: ```suggestion
These four implementation templates show what a fully implemented access strategy looks like, including which mechanisms are in use, how roles and teams are structured, and how the layers work together. They represent common organization archetypes, not specific companies. Use them as starting point for your own implementation.
```

> ## Choosing your template

Reviewer: ```suggestion
## Choose your template
```

Most enterprises are a blend of these archetypes. Use the following questions to identify which patterns are most relevant to you:

> | Question | If yes, look at... |

Reviewer: I think the format might be easier to read if the templates were formatted as a list under the "If yes, look at..." column. And maybe if the questions were bolded. Play around with the formatting and see if works better or not.

| :---- | :---- |
| Do you have regulated data that must be invisible to certain user groups? | [Template 1 (Financial)](#template-1-large-financial-institution) or [Template 2 (Regulated Multi-BU)](#template-2-regulated-enterprise-with-multiple-business-units) |
| Do you have hard compliance boundaries requiring complete data isolation between divisions? | [Template 2 (Regulated Multi-BU)](#template-2-regulated-enterprise-with-multiple-business-units) |
| Do you have high API key volume and many automation pipelines? | [Template 3 (Large Tech Company)](#template-3-large-technology-company) |
| Do you serve multiple internal tenants or clients on a shared platform? | [Template 4 (Platform Provider)](#template-4-government-agency--platform-provider) |
| Do you manage configuration through Terraform or similar IaC tools? | [Template 4 (Platform Provider)](#template-4-government-agency--platform-provider) for the most comprehensive IaC pattern |
| Do you operate many orgs and need consistent access controls across them? | [Template 4 (Platform Provider)](#template-4-government-agency--platform-provider) |

> ---

Reviewer: I would remove the divider lines unless they were added here intentionally.

| **Keys and tokens** | All keys managed through Terraform. Defense org uses a hardened Terraform pipeline with approval gates for any key or role change. Service accounts used exclusively. No human-owned application keys. |

### Key takeaway

Multi-org is justified here because of hard compliance boundaries. Regulations create strict data isolation and residency requirements. The defense division's use of Strict Mode reflects the default-deny requirement of its regulatory environment. Cross-Org Visibility keeps centralized reporting functional without compromising isolation.

---

## Template 3: Large Technology Company

### Profile

A global technology company with 12,000 Datadog users running a large-scale payment processing and commerce platform. Operates a single org representing 20 major product lines. Uses a proprietary identity system integrated with the Teams API and Terraform. Heavy API usage with over 500 active application keys and 200 service accounts.

### Access strategy

| Layer | Implementation |
| :---- | :---- |

> | **Org structure** | Single org representing 20 major product lines. Data Access Control and Teams provide internal boundaries within each org. 2 additional orgs used for sandbox and non-prod environments. |

Reviewer: ```suggestion
| **Org structure** | Single org representing 20 major product lines. Data Access Control and Teams provide internal boundaries within each org. Two additional orgs used for sandbox and non-prod environments. |
```

| **Asset protections** | Each agency's Monitors and Dashboards restricted to their Team for Edit access. Critical shared infrastructure Monitors (network, DNS, shared compute) restricted to the Platform team. An admin override team ("Platform Governance") is included in all asset access lists to prevent lockout. |
| **Keys and tokens** | All API keys and service accounts managed through Terraform with a ServiceNow approval gate. One API key per agency for data submission. Service accounts for shared automation pipelines. No human application keys permitted. All API access goes through service accounts with SATs. |
| **Managing as code** | All roles, Teams, Data Access Control datasets, granular access policies, and keys defined in Terraform. Changes go through code review and ServiceNow approval before being applied. This is essential at this scale: manual configuration across 200 tenant boundaries would be unsustainable. |

### Key takeaway

> Within-org segmentation (Teams + Data Access Control) can replace multi-org isolation when the tenant boundaries are organizational, not regulatory. The CMDB-driven identity chain and Terraform-managed configuration are not optional at this scale. The admin override team is a critical safety net that should be established before any asset restrictions are applied.

Reviewer: ```suggestion
Within-org segmentation (Teams + Data Access Control) can replace multi-org isolation when the tenant boundaries are organizational, not regulatory. The CMDB-driven identity chain and Terraform-managed configuration are required at this scale. The admin override team is a critical safety net that should be established before any asset restrictions are applied.
```
